src/test/java/org/folio/integration/http/VertxOkapiHttpClientTest.java

```diff
@@ -54,6 +59,30 @@ void afterEach() {
     stopVertx();
   }
 
+  @SneakyThrows
+  @Test
+  void throwsTimeoutWhenResponseTakesOverFiveSeconds() {
+    final var getEndpoint = matchingFolioHeaders(get(urlPathEqualTo("/record")));
+
+    fakeWebServer.stubFor(getEndpoint.willReturn(ok()
+      .withFixedDelay(6000)
+      .withBody(dummyJsonResponseBody())
+      .withHeader("Content-Type", "application/json")));
+
+    final var client = createClient();
+
+    final var getCompleted = client.get(
+      "/record", Headers.toMap(fakeWebServer.baseUrl()));
+    try {
+      final var response = getCompleted.get(6, SECONDS);
+      assertThat("Timeout exception should have been thrown.", response, is(-1));
+    } catch(Exception e) {
+      String err = "The timeout period of 5000ms has been exceeded";
+      assertThat(e.getMessage().indexOf(err), not(-1));
+    }
```

Comment:
I think the intended behaviour being demonstrated by this test is an exception is thrown.

If that is the case, it might be possible to express that more explicitly by `assertThrows` from JUnit 5 rather than this mechanism.